presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,212 @@
+                            " WHERE query = '%s' ORDER BY created DESC LIMIT 1", sqlStatement)).row(0).get(0);
+                    return (new CancelQuery(queryId, serverAddress)).cancel();
+                }
+            }
+            return null;
+        }
+    }
+
+    private void doCancelTest(String sql)
+    {
+        Response response = null;
+        Boolean isQueryCancelled;
+
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+        Future<Boolean> executeQueryThread = executor.submit(new SqlExecuteCallable(sql));
+        Future<Response> cancelQueryThread = executor.submit(new SqlCancelCallable(sql, 10000, 100));
+
+        try {
+            isQueryCancelled = executeQueryThread.get(2, TimeUnit.MINUTES);
+        }
+        catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException("error in query execution thread", e);
+        }
+        catch (TimeoutException e) {
+            executeQueryThread.cancel(true);
+            throw new RuntimeException("query execution timelimit exceeded", e);
+        }
+
+        String queryState = (String) query(format("SELECT state from system.runtime.queries" +
+                " WHERE query = '%s' ORDER BY created DESC LIMIT 1", sql)).row(0).get(0);
+        if (!isQueryCancelled || !queryState.equalsIgnoreCase("failed")) {
+            throw new RuntimeException("failed to cancel the query");
+        }
+
+        try {
+            response = cancelQueryThread.get();
+        }
+        catch (Exception e) {
+            throw new RuntimeException("error in cancel query thread", e);
+        }
+
+        if (response == null) {
+            throw new RuntimeException("error in cancel query thread");
+        }
+
+        if (response.getStatusCode() != HttpStatus.OK.code() && response.getStatusCode() != HttpStatus.ACCEPTED.code() &&
+                response.getStatusCode() != HttpStatus.NO_CONTENT.code()) {
+            throw new RuntimeException("unexpected error code " + response.getStatusCode() + "; reason=" + response.getStatusMessage());
+        }
+    }
+
+    @Test(groups = {CANCEL_QUERY})
+    public void cancelCreateTableTest()
+            throws SQLException
+    {
+        String sql = format("CREATE TABLE %s AS " +
+                "SELECT orderkey, partkey, shipinstruct FROM tpch.sf1.lineitem", tableName);
+
+        doCancelTest(sql);
+
+        try {
+            query(format("DROP TABLE %s", tableName));
+            throw new RuntimeException("cancelCreateTableTest failed");
+        }
+        catch (QueryExecutionException exceptionDueToNonExistentTableName) {
+        }
+    }
+
+    @Test(groups = {CANCEL_QUERY})
+    public void cancelInsertIntoTest()
+            throws SQLException
+    {
+        query(format("CREATE TABLE %s " +
+                "(orderkey BIGINT, partkey BIGINT, shipinstruct VARCHAR(25)) ", tableName));
+        query(format("INSERT INTO %s " +
```

Comment:
To make sure that any previous data in the table is not modified by INSERT INTO command.